r/aws_api_gateway_rest_api: Crash when `binary_media_types = [""]`

Terraform AWS Provider **v5.1.0**:

```hcl
resource "aws_api_gateway_rest_api" "example" {
  body = jsonencode({
    openapi = "3.0.1"
    info = {
      title   = "example"
      version = "1.0"
    }
    paths = {
      "/path1" = {
        get = {
          x-amazon-apigateway-integration = {
            httpMethod           = "GET"
            payloadFormatVersion = "1.0"
            type                 = "HTTP_PROXY"
            uri                  = "https://ip-ranges.amazonaws.com/ip-ranges.json"
          }
        }
      }
    }
  })

  name = "example"

  binary_media_types = [""]
  endpoint_configuration {
    types = ["REGIONAL"]
  }
}
```

Crash:

```
panic: interface conversion: interface {} is nil, not string

goroutine 483 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/apigateway.resourceRestAPIUpdate({0xf139e40, 0xc006c847b0}, 0xc007fbfa80, {0xdc1b800?, 0xc00038b800?})
  github.com/hashicorp/terraform-provider-aws/internal/service/apigateway/rest_api.go:409+0x2f74
github.com/hashicorp/terraform-provider-aws/internal/provider.interceptedHandler[...].func1(0x0?, {0xdc1b800?, 0xc00038b800?})
  github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:108 +0x178
...
Error: The terraform-provider-aws_v5.1.0_x5 plugin crashed!
```

--- a/.changelog/32169.txt
+++ b/.changelog/32169.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_api_gateway_rest_api: Fix crash when `binary_media_types` is `null`
+```

--- a/internal/service/apigateway/rest_api.go
+++ b/internal/service/apigateway/rest_api.go
@@ -395,19 +395,23 @@ func resourceRestAPIUpdate(ctx context.Context, d *schema.ResourceData, meta int
 			// Remove every binary media types. Simpler to remove and add new ones,
 			// since there are no replacings.
 			for _, v := range old {
-				operations = append(operations, &apigateway.PatchOperation{
-					Op:   aws.String(apigateway.OpRemove),
-					Path: aws.String(fmt.Sprintf("/%s/%s", prefix, escapeJSONPointer(v.(string)))),
-				})
+				if e, ok := v.(string); ok {
+					operations = append(operations, &apigateway.PatchOperation{
+						Op:   aws.String(apigateway.OpRemove),
+						Path: aws.String(fmt.Sprintf("/%s/%s", prefix, escapeJSONPointer(e))),
+					})
+				}
 			}
 
 			// Handle additions
 			if len(new) > 0 {
 				for _, v := range new {
-					operations = append(operations, &apigateway.PatchOperation{
-						Op:   aws.String(apigateway.OpAdd),
-						Path: aws.String(fmt.Sprintf("/%s/%s", prefix, escapeJSONPointer(v.(string)))),
-					})
+					if e, ok := v.(string); ok {
+						operations = append(operations, &apigateway.PatchOperation{
+							Op:   aws.String(apigateway.OpAdd),
+							Path: aws.String(fmt.Sprintf("/%s/%s", prefix, escapeJSONPointer(e))),
+						})
+					}
 				}
 			}
 		}
@@ -626,18 +630,22 @@ func resourceRestAPIWithBodyUpdateOperations(d *schema.ResourceData, output *api
 	}
 
 	if v, ok := d.GetOk("binary_media_types"); ok && len(v.([]interface{})) > 0 {
-		for _, elem := range aws.StringValueSlice(output.BinaryMediaTypes) {
-			operations = append(operations, &apigateway.PatchOperation{
-				Op:   aws.String(apigateway.OpRemove),
-				Path: aws.String("/binaryMediaTypes/" + escapeJSONPointer(elem)),
-			})
+		if len(output.BinaryMediaTypes) > 0 {
+			for _, elem := range aws.StringValueSlice(output.BinaryMediaTypes) {
+				operations = append(operations, &apigateway.PatchOperation{
+					Op:   aws.String(apigateway.OpRemove),
+					Path: aws.String("/binaryMediaTypes/" + escapeJSONPointer(elem)),
+				})
+			}
 		}
 
 		for _, elem := range v.([]interface{}) {
-			operations = append(operations, &apigateway.PatchOperation{
-				Op:   aws.String(apigateway.OpAdd),
-				Path: aws.String("/binaryMediaTypes/" + escapeJSONPointer(elem.(string))),
-			})
+			if el, ok := elem.(string); ok {
+				operations = append(operations, &apigateway.PatchOperation{
+					Op:   aws.String(apigateway.OpAdd),
+					Path: aws.String("/binaryMediaTypes/" + escapeJSONPointer(el)),
+				})
+			}
 		}
 	}
 

